src/SlmQueueSqs/Queue/SqsQueue.php

```diff
@@ -64,6 +66,15 @@ public function push(JobInterface $job, array $options = array())
             'DelaySeconds' => isset($options['delay_seconds']) ? $options['delay_seconds'] : null
         );
 
+        if ($this->isFifoQueue()) {
+            if (!isset($options['message_group_id']) || empty($options['message_group_id'])) {
+                throw new Exception\MissingMessageGroupException();
+            }
+
+            $parameters['MessageGroupId'] = $options['message_group_id'];
+            $parameters['MessageDeduplicationId'] = md5($parameters['MessageBody']);
```

Comment:
You can already do by default a content-based deduplication only with a setting in AWS thus not requiring that MessageDeduplicationId. 

My opinion is that we should let the user decide if they set the deduplication id because:
- MessageDeduplicationId might not be required if content-based deduplication is enabled for the queue
- Users might want to enforce their deduplication-id-generation technique

Reply:
I agree with leaving the choice to the user as I did for the `MessageGroupId`.

Reply:
Cheers! What about batch push?

Reply:
I just pushed a commit that also add the parameters to the `batchPush` method.